### DISCUSSION TITLE
Dialog disable escape key behaviour ?

### DISCUSSION THREAD

**Discussion by ElBambu** [tailwindlabs/headlessui#2367] [Help]
Hello is there a way to disable the escape key behaviour ? 

If i put a stop propagation on the Dialog or the Dialog.Panel it seems to be working but if i focus an element inside the Dialog.panel and then press the escape key it closes the Dialog :V

**Comment by victorrik** [comment]
im using this for now, but im not sure if is the best way

``` javascript

  useEffect(() => {
    addEventListener('keydown', handleKey, false);
    return () => {
      removeEventListener('keydown', handleKey, false);
    };
  }, [open]);

  const handleKey = (event: KeyboardEvent) => {
    if (event.key === 'Escape' && open) {
      event.preventDefault();
    }
  };

```
